File: leetcode-submissions/1710_Maximum_Units_on_a_Truck.ts
function maximumUnits(boxTypes: number[][], truckSize: number): number {
    boxTypes.sort((boxA,boxB)=>boxB[1]-boxA[1]);
    // after sorting we need to use the exact concept of the fractional knapsack 
   let ans :number =0;

   for(let i:number =0;i<boxTypes.length;i++){
    if(truckSize>=boxTypes[i][0]){
        truckSize-=boxTypes[i][0];
        ans +=(boxTypes[i][0]*boxTypes[i][1]);
    }else {
        // if we have not a sufficient size then we need to take a part of it according to the remaining trucksize 
        ans+=truckSize*boxTypes[i][1];
        return ans ;
    }
   }
   return ans  ;
};

File: leetcode-submissions/1710_Maximum_Units_on_a_Truck.cpp
class Solution {
public:


    int maximumUnits(vector<vector<int>>& boxTypes, int truckSize) {
       sort(boxTypes.begin(),boxTypes.end(),[] (auto &a,auto &b){return a[1]>b[1];});
           
    int ans=0;
        for(int i=0;i<boxTypes.size();i++){
            if(boxTypes[i][0]<=truckSize){
                ans+=boxTypes[i][1]*boxTypes[i][0];
                truckSize=truckSize-boxTypes[i][0];
            }else{
                ans+=boxTypes[i][1]*truckSize;
                truckSize=0;
            }
        }
        return ans;
}
    
};